src/repository.rs

```
@@ -1,71 +1,67 @@
 use super::debug;
-use super::error;
+use super::error::Error;
 use super::package::get_local_dir;
 use std::collections::HashMap;
+use std::fs::File;
+use std::io;
 use std::io::{Read, Write};
 use std::path::Path;
 
-pub fn update_repository_cache() -> Result<(), error::Error> {
-    let uri = format!(
-        "https://raw.githubusercontent.com/Camerooooon/yfin-official-repository/master/packages"
-    );
-    let mut req = reqwest::get(&uri).unwrap();
+const YFIN_OFFICIAL_REPOSITORY: &str =
+    "https://raw.githubusercontent.com/Camerooooon/yfin-official-repository/master/packages";
 
-    if !req.status().is_success() {
-        return Err(error::Error::RequestFailed);
-    }
+pub fn update_repository_cache() -> Result<(), io::Error> {
+    let mut req = match reqwest::get(YFIN_OFFICIAL_REPOSITORY) {
+        Ok(r) => r,
+        Err(..) => std::process::exit(1),
```

Like I said in my previous review std::process::exit(1) doesn't actually tell the user what is going on and completely ignores the error handling provided by reqwest (at this point just unwrap it and panic if there is an error)